### PR TITLE
localization publishes odom to target_frame transform

### PR DESCRIPTION
**Pull request by jrgnicho**
This PR addresses the following:
The sicknav350 node publishes a transform from the "odom" frame to the user defined "target_frame_id".  The name for the target frame can be passed to the node as a ros parameter.

### CODE REVIEW

File: ros/sicknav350/sicknav350.cpp [hunk 1]
```diff
@@ -238,6 +244,24 @@ int main(int argc, char *argv[])
...
+    }
+    else
+    {
+      try
+      {
+        tf_listerner.lookupTransform(target_frame_id,sick_frame_id,ros::Time(0),sickn350_to_target_tf);
+      }
+      catch(tf::LookupException &exp)
+      {
+        ROS_ERROR_STREAM("Transform lookup between "<<sick_frame_id<<" and "<<target_frame_id<<" failed, exiting");
```
Comment (shaun-edwards) [inline ros/sicknav350/sicknav350.cpp:273]: Is there a race condition on tf?  What if the tf publishing node hasn't come up yet?  Should we wait longer and print a warning?

Comment (jrgnicho) [inline ros/sicknav350/sicknav350.cpp:273]: Yes, a race condition could take place, I'll add a waitFor... call.

**Comment by jrgnicho** [comment]
@shaun-edwards  the last commit should address the issue with the tf race condition

**Comment by shaun-edwards** [comment]
+1

**Pull request merged**